tests/com.espressif.idf.ui.test/src/com/espressif/idf/ui/test/operations/ProjectTestOperations.java

```diff
@@ -97,69 +103,136 @@ public static void createDebugConfiguration(String projectName, SWTWorkbenchBot
+						e.printStackTrace();
+					}
+				}
+			}
+			throw new RuntimeException("The 'idf_component.yml' file was not found in the 'main' subfolder.");
 		}
 	}
-	
-	public static void openProjectComponentYMLFileInTextEditorUsingContextMenu(String projectName, SWTWorkbenchBot bot) 
+
+	public static void findProjectCleanedFilesInBuildFolder(String projectName, SWTWorkbenchBot bot)
 	{
-	    SWTBotTreeItem projectItem = fetchProjectFromProjectExplorer(projectName, bot);
-	    if (projectItem != null) {
-	        projectItem.select();
-	        projectItem.expand();
-	        projectItem.getNode("main").expand();
-
-	        int maxAttempts = 2;
-	        for (int attempt = 0; attempt <= maxAttempts; attempt++) {
-	            SWTBotTreeItem fileToOpenItem = findTreeItem(projectItem.getNode("main"), "idf_component.yml");
-
-	            if (fileToOpenItem != null) {
-	            	fileToOpenItem.select();
-	            	fileToOpenItem.doubleClick();
-	                return;
-	            }
-
-	            else {
-	                try {
-	                    Thread.sleep(3000);
-	                } catch (InterruptedException e) {
-	                    e.printStackTrace();
-	                }
-	            }
-	        }
-	        throw new RuntimeException("The 'idf_component.yml' file was not found in the 'main' subfolder.");
-	    }
+		SWTBotTreeItem projectItem = fetchProjectFromProjectExplorer(projectName, bot);
+		if (projectItem != null)
+		{
+			projectItem.select();
+			projectItem.expand();
+			projectItem.getNode("build").expand();
+
+			boolean fileToBeAbsentItem = isFileAbsent(projectItem.getNode("build"), ".elf");
+			SWTBotTreeItem fileToFindItem = findTreeItem(projectItem.getNode("build"), "bootloader");
+			if (fileToFindItem != null && fileToBeAbsentItem != false)
+			{
+				return;
+			}
+			throw new RuntimeException("The files was not deleted from the 'build' subfolder..");
+		}
 	}
 
-	private static SWTBotTreeItem findTreeItem(SWTBotTreeItem parent, String itemName) {
-	    for (SWTBotTreeItem child : parent.getItems()) {
-	        if (child.getText().equals(itemName)) {
-	            return child;
-	        }
-	        SWTBotTreeItem found = findTreeItem(child, itemName);
-	        if (found != null) {
-	            return found;
-	        }
-	    }
-	    return null;
+	public static void findProjectFullCleanedFilesInBuildFolder(String projectName, SWTWorkbenchBot bot)
+	{
+		SWTBotTreeItem projectItem = fetchProjectFromProjectExplorer(projectName, bot);
+		if (projectItem != null)
+		{
+			projectItem.select();
+			projectItem.expand();
+			projectItem.getNode("build").expand();
+
+			boolean fileToBeAbsentItem = isFileAbsent(projectItem.getNode("build"), ".elf");
+			boolean fileToBeAbsentItem1 = isFileAbsent(projectItem.getNode("build"), "bootloader");
+			if (fileToBeAbsentItem != false && fileToBeAbsentItem1 != false)
+			{
+				return;
+			}
+			throw new RuntimeException("The 'build' subfolder was not fully cleaned.");
```

Comment:
I think instead of an exception a simple false return here is better option the failure must be in tests via check through assertion exception should be internal in tests I believe